File: Subsets/Subsets.java
class Solution {
    public List<List<Integer>> subsets;
    public int[] nums;
    public List<List<Integer>> subsets(int[] nums) {
        subsets = new ArrayList<>();
        this.nums = nums;
        
        List<Integer> subset = new ArrayList<>();
        backtrack(0, subset);
        return this.subsets;
    }
    
    public void backtrack(int i, List<Integer> subset) {
        if (i >= nums.length) {
            List<Integer> ans = new ArrayList<>();
            for(int num: subset) {
                ans.add(num);
            }
            subsets.add(ans);
        }
        else {
            
            backtrack(i + 1, subset);
            
            subset.add(nums[i]);
            backtrack(i + 1, subset);
            subset.remove(subset.size() - 1);
        }
    }
}

File: Subsets/Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.nums = nums
        self.subset = []
        self.subsets = []
        
        self.backtrack(0)
        
        return self.subsets
    
    def backtrack(self, i):
        if (i == len(self.nums)):
            ans = []
            for num in self.subset:
                ans.append(num)
            self.subsets.append(ans)
        else:
            self.backtrack(i + 1)
            
            self.subset.append(self.nums[i])
            self.backtrack(i + 1)
            self.subset.pop()